src/fetch_foursquare_info.py

```diff
@@ -0,0 +1,129 @@
+import os.path
+import requests
+import numpy as np
+import pandas as pd
+import re
+import json
+import datetime
+
+# Setting dataset paths
+REIMBURSEMENTS_DATASET_PATH = os.path.join('data', '2016-11-22-reimbursements.xz')
+COMPANIES_DATASET_PATH = os.path.join('data', '2016-09-03-companies.xz')
+FOURSQUARE_DATASET_PATH = os.path.join('data', 'foursquare-companies.json')
+
+# These are the keys to access Foursquare's API. You can create your own through the link bellow
+# https://pt.foursquare.com/developers/register
+CLIENT_ID = ''
+CLIENT_SECRET = ''
+# This is in YYYYMMDD format.
+VERSION = '20161021'
+
+# Required params to make a request to Foursquare's API
+DEFAULT_PARAMS = {'client_id': CLIENT_ID,
+                'client_secret': CLIENT_SECRET,
+                'v': VERSION}
+
+def load_cnpjs(subquota_description):
+    u_cols = ['cnpj_cpf', 'subquota_description']
+    docs = pd.read_csv(REIMBURSEMENTS_DATASET_PATH,
+                        low_memory=False,
+                        usecols=u_cols,
+                        dtype={'cnpj_cpf': np.str})
+    meals = docs[docs.subquota_description == subquota_description]
+    return meals['cnpj_cpf'].unique()
+
+def load_companies():
+    u_cols = ['cnpj', 'trade_name', 'zip_code', 'latitude', 'longitude']
+    all_companies = pd.read_csv(COMPANIES_DATASET_PATH,
+                                low_memory=False,
+                                usecols=u_cols,
+                                dtype={'trade_name': np.str})
+    all_companies['clean_cnpj'] = all_companies['cnpj'].map(cleanup_cnpj)
+    relevant_cnpjs = load_cnpjs('Congressperson meal')
+    return all_companies[all_companies['clean_cnpj'].isin(relevant_cnpjs)]
+
+def cleanup_cnpj(cnpj):
+    return re.sub("\D", "", cnpj)
+
+def remaining_companies(companies):
+    fetched_companies = load_foursquare_companies_dataset()
+    return companies[~companies['cnpj'].isin(fetched_companies['cnpj'])]
+
+def load_foursquare_companies_dataset():
+    if os.path.exists(FOURSQUARE_DATASET_PATH):
+        return pd.read_json(FOURSQUARE_DATASET_PATH)
+    else:
+        return pd.DataFrame(columns=['cnpj'])
+
+def get_venue(company):
+    venue = search(company)
+    if venue:
+        return fetch_venue_by_id(venue['id'])
+
+def search(company, timeout=5):
+    params = dict(DEFAULT_PARAMS)
+    params.update({
+        'query': company['trade_name'],
+        'll': '%s,%s' % (company['latitude'], company['longitude']),
+        'zip_code': company['zip_code'],
+        'intent': 'match'
+    })
+    url = 'https://api.foursquare.com/v2/venues/search'
+    response = requests.get(url, params=params)
+
+    return parse_search_results(response)
```

Comment:
I'd check response's HTTP status here before passing it to the parser (same for line 85). We should rely on the server header, not in the JSON contents to know how successful was the request.

Something like:

```python
def search(company, timeout=5):
    …
    response = requests.get(url, params=params)
    if 200 <= response.status_code < 300:
        return parse_search_results(response)
```